src/main/java/spoon/support/sniper/internal/AbstractSourceFragmentContextCollection.java

```diff
@@ -68,6 +73,17 @@ public boolean knowsHowToPrint(PrinterEvent event) {
 		throw new SpoonException("Unexpected PrintEvent: " + event.getClass());
 	}
 
+	private boolean checkIfPartOfRole(CtRole role) {
```

Comment:
Just reading the invocation of this method, I have no idea of what its intention is. Check if what is part of a role? What does it mean to be part of a role?

`check` is also usually reserved for methods that throw. For methods that are meant to return a boolean, I'd recommend starting with `is`, `has` etc.

Looking at the method body, this method returns true if the first `ElementSourceFragment` of `childFragments` matches the provided role. Otherwise it returns false. Can you explain those semantics? I also agree with @MartinWitt that the method needs refactoring, but first I want to try to understand its intent.